teuthology/report.py

```diff
@@ -157,7 +157,7 @@ def running_jobs_for_run(self, run_name):
         :returns:        A dict like: {'1': '/path/to/1', '2': 'path/to/2'}
         """
         jobs = self.jobs_for_run(run_name)
-        for job_id in jobs.keys():
+        for job_id in list(jobs):
```

Comment:
It is legal indeed, but there was another error message, which I don't remember exactly related to modifying dict_keys iterator during iteration, this is why we need to use `list(obj.keys())` or shorter form `list(obj)` in some places. I haven't separated from this change from the `instance` just because to avoid merge conflicts, it is overkill to separate each line.

Reply:
For the illustration compar:
```
Python 2.7.17 (default, Oct 24 2019, 12:43:41) [GCC] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> a=dict(x=1, y=2)
>>> for i in a.keys():
...    a.pop(i)
... 
2
1
```
and:
```
Python 3.7.3 (default, Apr 09 2019, 05:18:21) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a=dict(x=1, y=2)
>>> for i in a.keys():
...     a.pop(i)
... 
1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: dictionary changed size during iteration
```

Reply:
thanks! i didn't realized we are removing element from `jobs` while iterating through it!